docs/ECOS.md

```diff
@@ -91,6 +91,34 @@ As part of a regular config that EVE recieves from its controller, each ECO gets
         * dsId
         * siginfo
 
+EVE periodically pulls the configuration from the controller. The configuration specifies the desired end state. EVE implements "eventual consistentency" model. To reach the desired state from the current state of the system, EVE computes the required operations to be performed. EVE then executes those operations. At the end of each operation, EVE reports the state of the system back to the controller.
+
+The controller drives the ECO state transitions via the configuration. These state transitions are described below. Due to the eventual consistency model, a new configuration may result in zero or more state transitions for a given ECO.
+
+* Prepare an ECO
+  * The ECI(s) is transfer'ed to EVE's storage area and resources required for the ECO are reserved.
+  * EVE performs the operations to "prepare an ECO" for each entry in the array of ECO's in the new configuration and not present in older configuration.
+
+* Start an ECO
+  * A previously stopped ECO or an ECO that has never been started, will be started. At the end of this, ECO will transition to Running state. For an ECO which is already Running, this is a no-op.
+  * EVE performs the operation if the 'activate' flag is set to 'true' in the configuration.
+
+* Stop an ECO
+  * An ECO which is either running or transitioning to running state will be stopped. However, the mutated run time state is preserved. At the end of this stage, ECO will transiton to Stopped state. A subsequent start of the ECO will start the ECO with the previously mutated runtime state.
+  * EVE performs the operation if the 'activate' flag is set to 'false' in the configuration.
+
+* Purge an ECO
+  * An ECO which is either running or transitioning to running state will be stopped and the mutated run time state of the ECO is deleted. A subsequent action to start the ECO will start the ECO with a pristine runtime state.
+  * EVE performs this operation if the 'purge' counter in the configuration is greater than the 'purge' counter in the previous configuration.
```

Comment:
So a `Stop` is _not_ a pre-requisite to a `Purge`? This doesn't make sense to me. There should be clear state transitions.